Lib/test/test_clinic.py

```diff
@@ -2431,6 +2432,19 @@ class ClinicFunctionalTest(unittest.TestCase):
     locals().update((name, getattr(ac_tester, name))
                     for name in dir(ac_tester) if name.startswith('test_'))
 
+    def check_depr_star(self, pnames, fn, *args, **kwds):
+        regex = (
+            fr"Passing( more than)?( [0-9]+)? positional argument(s)? to "
+            fr"{fn.__name__}\(\) is deprecated. Parameter(s)? {pnames} will "
+            fr"become( a)? keyword-only parameter(s)? in Python 3.14"
+        )
+        with self.assertWarnsRegex(DeprecationWarning, regex) as cm:
+            # Record the line number, so we're sure we've got the correct stack
+            # level on the deprecation warning.
+            _, lineno = fn(*args, **kwds), sys._getframe().f_lineno
```

Comment:
I thought to use the relative to `check_depr_star.__code__.co_firstlineno`, but this should work too. At least if `sys._getframe()` works.

Reply:
Both variants are used in Lib/test. I think either is fine.